BrowserKit/Sources/ComponentLibrary/Buttons/PrimaryRoundedButton.swift

```diff
@@ -20,21 +20,12 @@ public class PrimaryRoundedButton: ResizableButton, ThemeApplicable {
         )
     }
 
-    private var highlightedTintColor: UIColor!
-    private var normalTintColor: UIColor!
-    private var foregroundColorForState: ((UIControl.State) -> UIColor)?
-
-    override open var isHighlighted: Bool {
-        didSet {
-            backgroundColor = isHighlighted ? highlightedTintColor : normalTintColor
-        }
-    }
+    private var backgroundColorForState: ((UIControl.State) -> UIColor)?
```

Comment:
Why save the color inside a `((UIControl.State) -> UIColor)` ? Can we have a variable for each color that we want to save instead? Otherwise feels kinds of cryptic to know which color/state we have in this class

Reply:
I think it could be nice to have a variable for each background and foreground colors we want to use and set the adequate color in `updateConfiguration` rather that using those closure properties. Then in `applyTheme` we could just call `setNeedsUpdateConfiguration()` to trigger a configuration update once the various colors are set rather than directly updating the configuration itself in multiple places.

What do you think? 
cc: @cyndichin

Reply:
I've made a proposal for the implementation I've suggested [here](https://github.com/mozilla-mobile/firefox-ios/pull/17741/commits/7710af7af7aab5d12bb0abcc05dfa46f1f37e62f) if you'd like to give it a look :)

Reply:
I've took the liberty to make a proposal of the approach I've given above on [this PR](https://github.com/mozilla-mobile/firefox-ios/pull/17741) (see latest changes). The idea is to set every property that won't be updated with state changes in `configure`, then in `updateConfiguration` we can just switch on the button's state and use the proper colors for background and foreground colors :) Let me know what you think @thatswinnie @lmarceau @cyndichin